### PR TITLE
Arc 1302 yarn replace

### PR DESCRIPTION
**Pull request by joshkay10**
Replaced NPM with yarn.

~~don't forget to delete NODE_MODULES first~~

### CODE REVIEW

**Review by bgvozdev — APPROVED** [review]

**Comment by mboudreau** [comment]
I've just tried it locally.  No need to delete `node_modules` folder from what I could see.

I also added yarn as part of the `engines` and `volta` configuration of package.json.  Just needs the yarnrc file and I think we're good to go :)

File: package.json [hunk 1]
```diff
@@ -134,6 +134,7 @@
 		"yaml-lint": "^1.2.4"
 	},
 	"volta": {
-		"node": "14.17.0"
```
Comment (joshkay10) [inline package.json:137]: NICE!!

**Comment by atrigueiro** [comment]
Looks good to me. We also need to make changes in our deployment pipelines, should probably try and deploy this to ddev/staging first before merging, otherwise we'll break our prod deployment as well.

File: src/config/.yarnrc [hunk 2]
```diff
@@ -0,0 +1,2 @@
+registry "https://registry.npmjs.org/"
+"@atlassian:registry" "https://registry.npmjs.org/"
```
Comment (mboudreau) [inline src/config/.yarnrc:2]: this needs to be at the top level of the project.

**Review by mboudreau — CHANGES_REQUESTED** [review]

File: src/config/.yarnrc (hunk 2 above)
Comment (joshkay10) [reply src/config/.yarnrc]: moved, good catch!

**Review by bgvozdev — DISMISSED** [review]

**Review by mboudreau — DISMISSED** [review]

File: .github/workflows/on-push.yml [hunk 3]
```diff
@@ -8,32 +8,36 @@ jobs:
...
-        uses: actions/cache@v1
+      - uses: actions/cache@v2
         with:
-          path: ./node_modules
-          key: ${{ runner.os }}-${{hashFiles('**/package-lock.json')}}-node_modules
-      - run: npm install
-      - run: npm run db:test
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - run: volta pin yarn@latest
```
Comment (mboudreau) [inline .github/workflows/on-push.yml:38]: I don't think you need to pin it.  You were missing yarn in volta config :)

File: .github/workflows/on-push.yml [hunk 4]
```diff
@@ -8,32 +8,36 @@ jobs:
...
+      - uses: actions/cache@v2
         with:
-          path: ./node_modules
-          key: ${{ runner.os }}-${{hashFiles('**/package-lock.json')}}-node_modules
-      - run: npm install
-      - run: npm run lint
-      - run: npm run build
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+      - run: volta pin yarn@latest
```
Comment (mboudreau) [inline .github/workflows/on-push.yml:19]: I don't think you need to pin it.  You were missing yarn in volta config :)

File: .github/workflows/on-push.yml [hunk 5]
```diff
@@ -8,32 +8,36 @@ jobs:
 
     steps:
       - uses: actions/checkout@v1
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
```
Comment (mboudreau) [inline .github/workflows/on-push.yml:13]: pretty sure volta takes care of that too

File: .github/workflows/on-push.yml (hunk 4 above)
Comment (joshkay10) [reply .github/workflows/on-push.yml:19]: 1 way to find out!

**Review by mboudreau — APPROVED** [review]

**Review by atrigueiro — APPROVED** [review]

**Pull request merged**